### ISSUE TITLE
Snippets disappear with `.` key

### ISSUE THREAD

**Issue by andrewplummer** [microsoft/vscode#145960]
Does this issue occur when all extensions are disabled?: Yes/No
Yes

- VS Code Version: 1.65.2 (Universal)
- OS Version: OSX Monterey 12.3 (21E230)

Steps to Reproduce:

HTML/JSX snippets that include a `.` will disappear after the character is input:

<img width="1009" alt="Screen Shot 2022-03-24 at 6 37 09 PM" src="https://user-images.githubusercontent.com/74646/159888009-23904138-9961-4167-a5e4-6eef04c620a3.png">

<img width="710" alt="Screen Shot 2022-03-24 at 6 37 15 PM" src="https://user-images.githubusercontent.com/74646/159888017-cc383a47-feac-4b24-95a7-c3a8e566e3af.png">

This used to work in previous versions but seems to have stopped working with a recent update. I have ruled out Emmet as I have snippets turned off and the issue happens even with the extension disabled.

**Comment by jrieken** [comment]
Yes, snippets are trigger characters that "restart" the current session. When that happens snippets must prefix match with the respective snippet character but that doesn't seem to be the case here

**Comment by jrieken** [comment]
changed with https://github.com/microsoft/vscode/commit/e7a320c1db1ee2928a5ee57af437b032e002927c

**Comment by jrieken** [comment]
This is why we ask for snippets again when seeing a foreign trigger character after all: https://github.com/microsoft/vscode/commit/81f8c8bfb2644dfc816e991c00eca784a04e3210

**Comment by andrewplummer** [comment]
Thanks for the super fast response!

I'm not quite sure what a "session" is here or what "asking for them again" implies but is there some way (any way) to configure to allow special characters in snippets? Is this a setup issue?

**Comment by andrewplummer** [comment]
No worries! I'm simply using different snippets for now... but to be clear is this a bug then? Or is it intended?

**Comment by jrieken** [comment]
Pushed a fix which will arrive with next insiders at the of this week. Fingers crossed that this doesn't unwanted side-effects, like too eager suggestions

**Comment by andrewplummer** [comment]
Excellent! Thank you

**Comment by jrieken** [comment]
Sorry, re-opening because this causes unwanted snippet completions...

**Comment by andrewplummer** [comment]
No worries, thank you for addressing (and so quickly). I have "workaround snippets" at the moment so it's not a pressing issue for me but great to know it's at least being considered.

**Comment by dmitrivMS** [comment]
Related: #139943
